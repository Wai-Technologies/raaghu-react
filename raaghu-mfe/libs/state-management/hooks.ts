import { TypedUseSelectorHook , useDispatch , useSelector } from "react-redux";
import type { RootState , AppDispatch } from "./index";

export const useAppSelector : TypedUseSelectorHook<RootState> = useSelector;
export const useAppDispatch = () => useDispatch<AppDispatch>();