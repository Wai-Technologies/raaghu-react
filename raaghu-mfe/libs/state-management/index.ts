import tagAdminReducer from "./tag-admin/tag-admin-slice";
import fileDescriptorsReducer from "./file-descriptors/file-descriptors-slice";
import directoryDescriptorsReducer from "./directory-descriptors/directory-descriptors-slice";
import auditLogsReducer from "./audit-logs/audit-logs-slice";
import languageTextsReducer from "./language-texts/language-texts-slice";
import pollAdminReducer from "./poll-admin/poll-admin-slice";
import planAdminReducer from "./plan-admin/plan-admin-slice";
import roleReducer from "./role/role-slice";
import organizationUnitReducer from "./organization-unit/organization-unit-slice";
import permissionsReducer from "./permissions/permissions-slice";
import scopesReducer from "./scopes/scopes-slice";
import featuresReducer from "./features/features-slice";
import { useDispatch } from "react-redux";
import { persistReducer, persistStore } from "redux-persist";
import { configureStore, combineReducers } from "@reduxjs/toolkit";
import storage from "redux-persist/lib/storage";
import languagesReducer from "./languages/languages-slice";
import PageNotFoundReducer from "./pagenotfound/pagenotfound-slice";
import formsResponseReducer from "./formsresponse/formsresponse-slice";
import HomeReducer from "./home/home-slice";
import logoutReducer from "./logout/logout-slice";
import registerReducer from "./register/register-slice";
import blogPostReducer from "./blog-post/blog-post-slice";
import pagesReducer from "./pages/pages-slice";
import { forgotPasswordReducer } from "./forgot-password/forgotpassword-slice";
import editionReducer from "./edition/edition-slice";
import { subscriptionReducer } from "./subscription/subscription-slice";
import languageTextReducer from "./language-text/language-text-slice";
import organizationReducer from "./organization-tree/organization-tree-slice";
import rolesReducer from "./role/role-slice";
import ClaimTypeReducer from "./claim-type/claim-type-slice";
import securityLogsReducer from "./security-logs/security-logs-slice";
import userReducer from "./user/user-slice";
import settingsReducer from "./settings/settings-slice";
import FileManagementReducer from "./file-management/file-management-slice";
import hostReducer from "./host/host-slice";
import applicationsReducer from "./applications/applications-slice";
import urlForwardingReducer from "./url-forwarding/url-forwarding-slice";
import textTemplateReducer from "./text-template/text-template-slice";
import tenantReducer from "./tenant/tenant-slice";
import chatsReducer from "./chats/chats-slice";
import bloggerReducer from "./blogger/blogger-slice";
import formsReducer from "./forms/forms-slice";
import paymentPlansReducer from "./payment-plans/paymentPlans-slice";
import personalDataReducer from "./personal-data/personal-data-slice";
import commentsReducer from "./comments/comments-slice";
import tagsReducer from "./tags/tags-slice";
import myAccountReducer from "./my-account/my-account-slice";
import paymentRequestsReducer from "./payment-requests/paymentRequests-slice";
import newslettersReducer from "./newsletters/newsletters-slice";
import pollsReducer from "./polls/polls-slice";
import globalResourcesReducer from "./global-resources/globalResources-slice";
import menusReducer from "./menus/menus-slice";
import blogsReducer from "./Blogs/blogs-slice";
import configureReducer from "./config/config-slice";

const persistConfig = {
    key: "root",
    storage,
    blacklist: ["forgotPassword"],
};
const rootReducer = combineReducers({
    tagAdmin: tagAdminReducer,
    fileDescriptors: fileDescriptorsReducer,
    directoryDescriptors: directoryDescriptorsReducer,
    auditLogs: auditLogsReducer,
    languageTexts: languageTextsReducer,
    pollAdmin: pollAdminReducer,
    planAdmin: planAdminReducer,
    role: roleReducer,
    organizationUnit: organizationUnitReducer,
    permissions: permissionsReducer,
    features: featuresReducer,
    languages: languagesReducer,
    configureStore: configureReducer,
    PageNotFound: PageNotFoundReducer,
    formsResponse: formsResponseReducer,
    Home: HomeReducer,
    Logout: logoutReducer,
    register: registerReducer,
    blogPost: blogPostReducer,
    pages: pagesReducer,
    forgotPassword: forgotPasswordReducer,
    subscription: subscriptionReducer,
    edition: editionReducer,
    languagesText: languageTextReducer,
    organization: organizationReducer,
    roles: rolesReducer,
    claimTypes: ClaimTypeReducer,
    securityLogs: securityLogsReducer,
    applications: applicationsReducer,
    scopes: scopesReducer,
    textTemplate: textTemplateReducer,
    auditLog: auditLogsReducer,
    user: userReducer,
    settings: settingsReducer,
    tenant: tenantReducer,
    chats: chatsReducer,
    blogger: bloggerReducer,
    forms: formsReducer,
    fileManagement: FileManagementReducer,
    paymentPlans: paymentPlansReducer,
    urlForwarding: urlForwardingReducer,
    personalData: personalDataReducer,
    comments: commentsReducer,
    tags: tagsReducer,
    host: hostReducer,
    myaccount: myAccountReducer,
    paymentRequests: paymentRequestsReducer,
    polls: pollsReducer,
    globalResources: globalResourcesReducer,
    newsletters: newslettersReducer,
    menus: menusReducer,
    blogs: blogsReducer
});
const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: { persistedReducer },
});
export const persistor = persistStore(store);
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
