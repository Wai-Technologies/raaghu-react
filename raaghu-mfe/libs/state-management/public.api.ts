export * from "./tag-admin/tag-admin-slice";
export * from "./file-descriptors/file-descriptors-slice";
export * from "./directory-descriptors/directory-descriptors-slice";
export * from "./audit-logs/audit-logs-slice";
export * from "./language-texts/language-texts-slice";
export * from "./poll-admin/poll-admin-slice";
export * from "./plan-admin/plan-admin-slice";
export * from "./role/role-slice";
export * from "./organization-unit/organization-unit-slice";
export * from "./permissions/permissions-slice";
export * from "./applications/applications-slice";
export * from "./scopes/scopes-slice";
export * from "./features/features-slice";
export * from "./edition/edition-slice";
export * from "./languages/languages-slice";
export * from "./pagenotfound/pagenotfound-slice";
export * from "./formsresponse/formsresponse-slice";
export * from "./home/home-slice";
export * from "./register/register-slice";
export * from "./blog-post/blog-post-slice";
export * from "./pages/pages-slice";
export * from "./index";
export * from "./forgot-password/forgotpassword-slice";
export * from "./subscription/subscription-slice";
export * from "./claim-type/claim-type-slice";
export * from "./security-logs/security-logs-slice";
export * from "./user/user-slice";
export * from "./language-text/language-text-slice";
export * from "./settings/settings-slice";
export * from "./tenant/tenant-slice";
export * from "./chats/chats-slice";
export * from "./file-management/file-management-slice";
export * from "./url-forwarding/url-forwarding-slice";
export * from "./host/host-slice";
export * from "./my-account/my-account-slice";
export * from "./forms/forms-slice";
export * from "./polls/polls-slice";
export * from "./newsletters/newsletters-slice";
export * from "./menus/menus-slice";
export * from "./Blogs/blogs-slice";
export * from "./logout/logout-slice";
export * from "./config/config-slice";
