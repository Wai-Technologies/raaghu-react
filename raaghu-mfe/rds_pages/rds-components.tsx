export * from "../../raaghu-components/src/index";
