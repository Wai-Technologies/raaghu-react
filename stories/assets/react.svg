<svg width="84" height="80" viewBox="0 0 84 80" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" fill="#61DAFB"/>
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M42 55.4964C64.0914 55.4964 82 48.5584 82 40C82 31.4415 64.0914 24.5035 42 24.5035C19.9086 24.5035 2 31.4415 2 40C2 48.5584 19.9086 55.4964 42 55.4964Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7734 47.7482C39.8191 67.1601 54.6952 79.4276 62 75.1484C69.3048 70.8692 66.2723 51.6637 55.2266 32.2518C44.1809 12.8398 29.3048 0.572317 22 4.85154C14.6952 9.13076 17.7277 28.3363 28.7734 47.7482Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7732 32.2518C17.7275 51.6637 14.695 70.8692 21.9998 75.1485C29.3046 79.4277 44.1807 67.1602 55.2264 47.7482C66.2721 28.3363 69.3046 9.1308 61.9998 4.85158C54.695 0.572351 39.8189 12.8398 28.7732 32.2518Z" stroke="#61DAFB" stroke-width="4"/>
</svg>
