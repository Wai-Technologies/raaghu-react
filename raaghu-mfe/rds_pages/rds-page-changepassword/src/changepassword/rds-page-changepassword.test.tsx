import React from "react";
import "@testing-library/jest-dom";
import { render, waitFor } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import ChangePassword from "./changepassword";
import { store, useAppDispatch } from "../../../../libs/state-management/index";
import { RdsCompChangeUserPassword } from "../../../rds-components";

jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
}));

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
        t: (key: string) => key,
    }),
}));

global.location = {
    pathname: "../../../../libs/proxy/core",
} as any;

// Mock the useAppDispatch function
jest.mock("../../../../libs/state-management/index.ts", () => ({
    ...jest.requireActual("../../../../libs/state-management/index.ts"),
    useAppDispatch: jest.fn(),
}));

// Mock the rds-elements and rds-components
jest.mock("../../../../../raaghu-elements/src", () => ({
    RdsButton: jest.fn(),
    RdsSearch: jest.fn(),
    RdsModal: jest.fn(),
    RdsInput: jest.fn()
}));

jest.mock("../../../../../raaghu-components/src", () => ({
    RdsCompChangeUserPassword: jest.fn(),
}));

describe("Change Password page", () => {
    it("renders Change Password page correctly", async () => {
        // Mock the useAppDispatch hook
        const mockDispatch = jest.fn();
        (useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);

        render(
            <Provider store={store}>
                <ChangePassword />
            </Provider>
        );
    });

    it("Check change passsword button click", async () => {
        const changePasswordData = jest.fn();
        const getPasswordDataSubmit = jest.fn();
        const mockDispatch = jest.fn();
        (useAppDispatch as jest.Mock).mockReturnValue(mockDispatch);

        const password = render(
            <Provider store={store}>
                <ChangePassword />
                <RdsCompChangeUserPassword changePasswordData={changePasswordData} PasswordDataSubmit={(data: any) => getPasswordDataSubmit(data)}/>
            </Provider>
        );
        expect(password).toMatchSnapshot();
    });

    afterEach(() => {
        jest.clearAllMocks();
        global.localStorage = {
            getItem: jest.fn(),
            setItem: jest.fn(),
            removeItem: jest.fn(),
            clear: jest.fn(),
            length: 0,
            key: jest.fn(),
        };
    });
});
