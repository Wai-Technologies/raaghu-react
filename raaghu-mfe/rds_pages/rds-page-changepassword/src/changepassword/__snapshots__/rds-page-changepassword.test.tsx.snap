// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Password page Check change passsword button click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-background"
      >
        <div
          class="align-items-center d-flex justify-content-center vh-100 m-auto login-container"
        >
          <div
            class="container-fluid m-2"
          >
            <div
              class="bg-white row rounded-3 "
            >
              <div
                class="col-md-6 align-items-center justify-content-center login m-auto"
              >
                <div
                  class="h-100 py-4 px-3"
                >
                  <div
                    class="py-4"
                  >
                    <div
                      class="text-center"
                    >
                      <img
                        src="./assets/raaghu_text_logo.svg"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="col-md-6 order-1 order-sm-2 rounded-end position-relative align-items-center p-0 login-card-height"
                style="background-image: url(../assets/bg_1.png); background-size: cover; background-position: bottom; background-repeat: no-repeat;"
              >
                <video
                  autoplay=""
                  class="video"
                  id="myVideo"
                  loop=""
                >
                  <source
                    src="../assets/Comp1.mp4"
                    type="video/mp4"
                  />
                </video>
                <div
                  class="raghu1"
                >
                  <img
                    src="../assets/fg_raaghu.png"
                  />
                </div>
                <div
                  class="wrap"
                >
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                  <div
                    class="c"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-background"
    >
      <div
        class="align-items-center d-flex justify-content-center vh-100 m-auto login-container"
      >
        <div
          class="container-fluid m-2"
        >
          <div
            class="bg-white row rounded-3 "
          >
            <div
              class="col-md-6 align-items-center justify-content-center login m-auto"
            >
              <div
                class="h-100 py-4 px-3"
              >
                <div
                  class="py-4"
                >
                  <div
                    class="text-center"
                  >
                    <img
                      src="./assets/raaghu_text_logo.svg"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="col-md-6 order-1 order-sm-2 rounded-end position-relative align-items-center p-0 login-card-height"
              style="background-image: url(../assets/bg_1.png); background-size: cover; background-position: bottom; background-repeat: no-repeat;"
            >
              <video
                autoplay=""
                class="video"
                id="myVideo"
                loop=""
              >
                <source
                  src="../assets/Comp1.mp4"
                  type="video/mp4"
                />
              </video>
              <div
                class="raghu1"
              >
                <img
                  src="../assets/fg_raaghu.png"
                />
              </div>
              <div
                class="wrap"
              >
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
